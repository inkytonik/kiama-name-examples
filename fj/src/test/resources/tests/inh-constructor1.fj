class A extends Object {
  Object f;
  A(Object f) {
    super();
    this.f = f;
  }
}
// Require that the parameter name is the same as field in supertype
class B extends A {
  B(Object g) {
    super(g);
  }
}
