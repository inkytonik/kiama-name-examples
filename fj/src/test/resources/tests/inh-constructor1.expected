src/test/resources/tests/inh-constructor1.fj:2:10:info: f def: Object
  Object f;
         ^
src/test/resources/tests/inh-constructor1.fj:3:12:info: f def: Object
  A(Object f) {
           ^
src/test/resources/tests/inh-constructor1.fj:10:3:error: constructor args (Object g) should be the fields (Object f)
  B(Object g) {
  ^
src/test/resources/tests/inh-constructor1.fj:10:12:info: g def: Object
  B(Object g) {
           ^
src/test/resources/tests/inh-constructor1.fj:11:11:error: super arg 0 must be f, not g
    super(g);
          ^
